libcob/call.c

```diff
@@ -289,6 +283,37 @@ static int last_entry_is_working_directory (const char *buff, const char *pstr)
 	return 0;
 }
 
+static void* cob_dlopen(const char* filename) {
+#ifdef _WIN32
+	if (filename == NULL) {
+		return GetModuleHandle (NULL);
+	}
+	return LoadLibrary(filename);
+#elif	defined(USE_LIBDL)
+	int flags = cobsetptr->cob_load_global 
+		? RTLD_LAZY | RTLD_GLOBAL
+		: RTLD_LAZY | RTLD_LOCAL;
+
+	return dlopen(filename, flags);
+#else
+    if (advise != NULL) {
```

Comment:
advise is global, so only set if the "last set" variant is different to the current one - you can store the last used cob_load_global as a static var within this function for that